core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/allowances/AbstractAllowanceWithRanges.java

```diff
@@ -190,10 +190,13 @@ private Envelope[] computeAllowanceRegion(Envelope envelopeRegion, EnvelopeSimCo
             var envelopeRange = Envelope.make(envelopeRegion.slice(range.beginPos, range.endPos));
             var imposedBeginSpeed = imposedTransitionSpeeds[rangeIndex];
             var imposedEndSpeed = imposedTransitionSpeeds[rangeIndex + 1];
+            var tolerance = context.timeStep * envelopeRange.getTotalTime() / envelopeRegion.getTotalTime();
             var allowanceRange =
-                    computeAllowanceRange(envelopeRange, context, range.value, imposedBeginSpeed, imposedEndSpeed);
+                    computeAllowanceRange(envelopeRange, context, range.value, imposedBeginSpeed, imposedEndSpeed,
+                            tolerance);
             // memorize the beginning and end speeds
-            imposedTransitionSpeeds[rangeIndex] = allowanceRange.getBeginSpeed();
+            imposedTransitionSpeeds[rangeIndex] =
+                    allowanceRange.getBeginSpeed(); // shouldn't we check if it's the same ?
```

Comment:
I guess we could assert the continuity of the transition speed yes, though it's being done at the end of the envelope build.